src/main.rs

```diff
@@ -76,6 +68,34 @@ use smp::PerCpu;
 use sync::Once;
 use umode::UmodeTask;
 
+#[cfg(test)]
+pub trait Testable {
+    fn run(&self) -> TestResult;
+}
+
+#[cfg(test)]
+impl<T> Testable for T
+where
+    T: Fn() -> TestResult,
+{
+    fn run(&self) -> TestResult {
```

Comment:
it would make it easier to write tests if `TestResult` were actually a result, because then you could use the `?` operator.

Reply:
done

Reply:
Think you pushed the wrong branch -- here's how it looks like for me: `fn run(&self) -> TestResult;` (instead of  `fn run(&self) -> Result<blah>;`

Reply:
TestResult is a type in test_system 
```
pub type TestResult = Result<(), TestFailure>;
```